### PR TITLE
Gusto OAuth

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

File: src/app/auth/login/login.component.ts [hunk 1]
```diff
@@ -32,6 +34,7 @@ export class LoginComponent implements OnInit {
       };
       this.userService.storeUserProfile(user);
       this.qbdAuthService.qbdLogin(user.refresh_token).subscribe();
+      this.orgService.createOrg().subscribe();
```
Comment (ashwin1111) [inline src/app/auth/login/login.component.ts:37]: why is this needed?

File: src/app/core/models/enum/enum.model.ts [hunk 2]
```diff
@@ -12,17 +12,20 @@ export enum AccountingIntegrationApp {
     XERO = 'Xero'
 }
 
+
 export enum InAppIntegration {
     BAMBOO_HR = 'Bamboo HR',
     QBD = 'QuickBooks Desktop',
-    TRAVELPERK = 'TravelPerk'
+    TRAVELPERK = 'TravelPerk',
+    GUSTO = 'GUSTO'
```
Comment (ashwin1111) [inline src/app/core/models/enum/enum.model.ts:20]: Gusto

File: src/app/integrations/bamboo-hr/bamboo-hr.component.html [hunk 3]
```diff
@@ -14,7 +14,7 @@
 </p-toast>
 
 <app-app-header [iconPath]="'assets/logos/bamboo-hr-logo.svg'" [appName]="'BambooHR'" [appDescription]="'Auto
-sync employee details between Bamboo HR and Fyle.'" [isLoading]="isLoading" [hideRefreshIcon]="hideRefreshIcon" [showErrorScreen]="showErrorScreen" [bambooHrConfiguration]="bambooHrConfiguration" [isIntegrationSetupInProgress]="isBambooSetupInProgress" [isIntegrationConnected]="isBambooConnected" (openDialog)="openDialog()" (disconnectBambooHr)="disconnectBambooHr()" (syncEmployees)="syncEmployees()"></app-app-header>
+sync employee details between Bamboo HR and Fyle.'" [isLoading]="isLoading" [hideRefreshIcon]="hideRefreshIcon" [showErrorScreen]="showErrorScreen" [integrationConfiguration]="bambooHrConfiguration" [isIntegrationSetupInProgress]="isBambooSetupInProgress" [isIntegrationConnected]="isBambooConnected" (openDialog)="openDialog()" (disconnectIntegration)="disconnectBambooHr()" (syncEmployees)="syncEmployees()"></app-app-header>
```
Comment (ashwin1111) [inline src/app/integrations/bamboo-hr/bamboo-hr.component.html:17]: write the string in a single line, `Auto sync employee details between Bamboo HR and Fyle.`

File: src/app/integrations/gusto/gusto.component.html [hunk 4]
```diff
@@ -1 +1,22 @@
-<p>gusto works!</p>
+<div>
+    <app-app-header [iconPath]="'assets/logos/gusto-logo.svg'" [appName]="'Gusto'" [appDescription]="'Auto sync employee details between Fyle and Gusto.'" [isLoading]="isLoading" [hideRefreshIcon]="hideRefreshIcon" [showErrorScreen]="showErrorScreen" [integrationConfiguration]="gustoConfiguration" [isIntegrationSetupInProgress]="isGustoSetupInProgress" [isIntegrationConnected]="isGustoConnected" (disconnectIntegration)="disconnectGusto()" (syncEmployees)="syncEmployees()"></app-app-header>
+
+    <app-app-integration-landing  *ngIf="!isLoading && !isGustoConnected && !showErrorScreen"  [headerText]="'Connect your Gusto account to your Fyle account to atuo-sync employee data.'" [svgPath]="'assets/flow-charts/gusto-flow-chart.svg'" [redirectLink]="RedirectLink.GUSTO"></app-app-integration-landing>
+
+    <app-error *ngIf="showErrorScreen"></app-error>
+
+    <!-- <app-configuration *ngIf="!isLoading && isGustoConnected"></app-configuration> -->
```
Comment (ashwin1111) [inline src/app/integrations/gusto/gusto.component.html:8]: why is this commented out?

File: src/app/integrations/gusto/gusto.component.html [hunk 5]
```diff
@@ -1 +1,22 @@
-<p>gusto works!</p>
+<div>
+    <app-app-header [iconPath]="'assets/logos/gusto-logo.svg'" [appName]="'Gusto'" [appDescription]="'Auto sync employee details between Fyle and Gusto.'" [isLoading]="isLoading" [hideRefreshIcon]="hideRefreshIcon" [showErrorScreen]="showErrorScreen" [integrationConfiguration]="gustoConfiguration" [isIntegrationSetupInProgress]="isGustoSetupInProgress" [isIntegrationConnected]="isGustoConnected" (disconnectIntegration)="disconnectGusto()" (syncEmployees)="syncEmployees()"></app-app-header>
```
Comment (ashwin1111) [inline src/app/integrations/gusto/gusto.component.html:2]: use enum for app name

Comment (ashwin1111) [reply src/app/integrations/gusto/gusto.component.html:2]: description string to be written in a single line

Comment (ashwin1111) [reply src/app/integrations/gusto/gusto.component.html:2]: when there are a lot if input/output, format the code better
```
<app-app-header
    [iconPath]="'assets/logos/gusto-logo.svg'"
    [appName]="'Gusto'"
    [appDescription]="'Auto sync employee details between Fyle and Gusto.'"
    [isLoading]="isLoading"
    [hideRefreshIcon]="hideRefreshIcon"
    [showErrorScreen]="showErrorScreen"
    [integrationConfiguration]="gustoConfiguration"
    [isIntegrationSetupInProgress]="isGustoSetupInProgress"
    [isIntegrationConnected]="isGustoConnected"
    (disconnectIntegration)="disconnectGusto()"
    (syncEmployees)="syncEmployees()"
></app-app-header>
```

Comment (ashwin1111) [reply src/app/integrations/gusto/gusto.component.html:2]: please do this everywhere in this PR

File: src/app/integrations/gusto/gusto.component.ts [hunk 6]
```diff
@@ -7,9 +16,141 @@ import { Component, OnInit } from '@angular/core';
...
+    this.toastService.displayToastMessage(ToastSeverity.SUCCESS, 'Syncing Employees Started');
+    this.gustoService.syncEmployees().subscribe(() => {
+      this.hideRefreshIcon = false;
+    });
+  }
+
+  disconnectGusto(): void {
+    this.trackingService.onClickEvent(ClickEvent.DISCONNECT_GUSTO);
+    this.isLoading = true;
+    // This.gustoService.disconnectBambooHr().subscribe(() => {
```
Comment (ashwin1111) [inline src/app/integrations/gusto/gusto.component.ts:67]: why is this commented?

File: src/app/auth/login/login.component.ts (hunk 1 above)
Comment (DhaaraniCIT) [reply src/app/auth/login/login.component.ts:37]: to get the manager id

Comment (ashwin1111) [reply src/app/auth/login/login.component.ts:37]: what is manager id? We don't do this in bamboo, how does it work there?

Comment (ashwin1111) [reply src/app/auth/login/login.component.ts:37]: also, this will be called in integration component as well

File: src/app/integrations/gusto/gusto.component.ts (hunk 6 above)
Comment (DhaaraniCIT) [reply src/app/integrations/gusto/gusto.component.ts:67]: api not yet developed
